From 508c0cc79e44dbafadde69c8d38a6e73de059cc4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 15:56:29 +0300
Subject: [PATCH] Revert "[SW-82716] common: add sync support for i2c"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit ea75bd4122c172e2762d9ea2e70b507174b5140f.

Change-Id: I27237a665d4e5414b5a4a192ba7f740463d9c379
---

diff --git a/common/module_fwupd.c b/common/module_fwupd.c
index 4dc3d18..fc02b2a 100644
--- a/common/module_fwupd.c
+++ b/common/module_fwupd.c
@@ -16,7 +16,6 @@
 #include <fit_load_type.h>
 #include <security.h>
 #include <memory.h>
-#include <delay.h>
 
 #include "cmd_module.h"
 #include "fit.h"
@@ -62,18 +61,6 @@
 static struct fwupd_state i2c_state;
 static struct fwupd_resp_prop resp_props[RESP_PROP_NUM];
 
-#define FWUPD_NORET_DELAY_US	200000
-
-static void fwupd_sync_exec(struct fwupd_state *state)
-{
-	if (state->sync_exec_res == NOT_FOUND || !state->noret_delay)
-		return;
-
-	if (arcboot_rom_timeout_expired(state->noret_delay, FWUPD_NORET_DELAY_US)) {
-		state->exec_res = state->sync_exec_res;
-		state->sync_exec_res = NOT_FOUND;
-	}
-}
 
 uint32_t i2c_fwupd_cmd_readl(void)
 {
@@ -345,7 +332,6 @@
 	if (state->sending_error_flag && arcboot_error_is_pending())
 		return FOUND_SLOW_RETURN;
 
-	fwupd_sync_exec(state);
 	if (state->exec_res != NOT_FOUND)
 		return state->exec_res;
 
@@ -534,13 +520,7 @@
 	if (err)
 		return err;
 
-	if (state->synced_no_return) {
-		state->sync_exec_res = FOUND_NO_RETURN;
-		state->noret_delay = arcboot_rom_get_ts_us();
-	} else {
-		state->exec_res = FOUND_NO_RETURN;
-	}
-
+	state->exec_res = FOUND_NO_RETURN;
 	return err;
 }
 
